tests/system/robot/symbolPronunciationTests.py

```diff
@@ -0,0 +1,640 @@
+	CARET_HOME = "control+home"
+	CARET_CHAR = "rightArrow"
+	CARET_CHAR_BACK = "leftArrow"
+
+
+class SymLevel(_enum.Enum):
+	"""Symbol levels, should match characterProcessing.SymbolLevel
+	"""
+	NONE = 0
+	ALL = 300
+
+
+class EndSpeech(_enum.Enum):
+	""" Speech given when reaching the end of the movement direction.
+	"""
+	BOTTOM = "Bottom"
+	RIGHT = "Right"
+	NONE = None
+
+
+def _pressKeyAndCollectSpeech(key: str, numberOfTimes: int) -> _typing.List[str]:
+	actual = []
+	for _ in range(numberOfTimes):
+		spoken = _NvdaLib.getSpeechAfterKey(key)
+		# collect all output before asserting to show full picture of behavior
+		actual.append(spoken)
+	return actual
+
+
+def _getByWordTestSample() -> str:
+	return (
+		"Test: "  # first word won't be spoken
+		'Say (quietly) "Hello, Jim ".'
+		" don't"  # test punctuation inside a word.
+		# symbols have space before and after, so it is considered a word
+		" ➔ 👕 \n"  # test Symbols containing punctuations (right-pointing arrow, t-shirt)
+		"1 | 2 || 3 ||| 4\n"  # test single/multiple symbols within a "word" issue #10855
+		' \n'  # single space
+		'\t\n'  # single tab
+		'    \n'  # 4 spaces
+		'➔\n'  # no space before, only newline after symbol
+		'👕\n'  # no space before, only newline after symbol
+		'👕'  # no space before, no newline after symbol
+	)
+
+
+def _getMoveByLineTestSample() -> str:
+	testData = [
+		"Test:",  # initial new line which isn't spoken
+		'Say',
+		'(quietly)',  # test parenthesis
+		'"Hello,',  # test quote, comma
+		'Jim".',  # test quote, dot
+		" don't ",  # test punctuation inside a word.
+		'➔', '👕',  # test Symbols containing punctuations (right-pointing arrow, t-shirt)
+		'➔ ', '👕 ',  # test Symbols containing punctuations with joined space
+		'➔👕',  # test Symbols containing punctuations without space
+		"1 | 2 || 3 ||| 4",  # test single/multiple symbols within a "word" issue #10855
+		' ',  # single space
+		'\t',  # single tab
+		'    ',  # 4 spaces
+		'',  # to ensure prior entry ends with newline.
+	]
+	return '\n'.join(testData)
+
+
+def _getMoveByCharTestSample() -> str:
+	# Intentionally concat the following strings, there should be no trailing commas
+	return (
+		'T'  # An initial character, that isn't spoken (because it isn't traversed during nav).
+		'S ()"'
+		"'e,➔👕\t"
+		'\na'  # Note: The 'a' character will be on the next line, thus won't be spoken
+	)
+
+
+def test_moveByWord():
+	"""Move by word with symbol level 'all' then with symbol level 'none'
+	Symbol expectations:
+	➔ - When replaced by speech "right-pointing arrow", the dash should not be removed.
+	👕 - When replaced by speech "t-shirt", the dash should not be removed, honor the replacement text for
+			symbols/punctuation.
+
+	There should not be any "empty" words. E.G. a word made of two bar chars: ||
+	Something should be reported, even at symbol level None. Possible:
+	- The names of the characters: "bar bar", what if there are many (hundreds) symbols?
```

Comment:
I'd say this is a solved problem since when there are more than three characters in a row we just report their count assuming that the character in question is spoken at the current symbol level. I see no reason why this should work differently when navigating by word.

Reply:
What if they are different characters, eg `!@#$%^`

Reply:
`@^*_` is a better example for notepad.exe because it is treated as a word.

Reply:
I don't see what is the problem here. If we decide to report the words at the symbol level all then each symbol should be reported to remain consistent with reporting when user sets symbol level to all manually.